src/vmaf.rs

```diff
@@ -12,11 +12,22 @@ pub fn run(
     reference: &Path,
     distorted: &Path,
     filter_complex: &str,
+    cuda: bool,
 ) -> anyhow::Result<impl Stream<Item = VmafOut>> {
+    if cuda {
+        eprintln!("\nDEBUG: Using ffmpeg -filter_complex {filter_complex}");
+    }
+
     let vmaf: ProcessChunkStream = Command::new("ffmpeg")
         .kill_on_drop(true)
+        .arg2_if(cuda, "-hwaccel", "cuda")
+        .arg2_if(cuda, "-hwaccel_output_format", "cuda")
+        .arg2_if(cuda, "-codec:v", "av1_cuvid") // TODO: determine this automatically?
         .arg2("-r", "24")
         .arg2("-i", distorted)
+        .arg2_if(cuda, "-hwaccel", "cuda")
+        .arg2_if(cuda, "-hwaccel_output_format", "cuda")
+        .arg2_if(cuda, "-codec:v", "av1_cuvid") // TODO: determine this automatically?
```

Comment:
Hardcoded -c:v "av1_cuvid" from https://ffmpeg.org/ffmpeg-filters.html#Examples-93. This presumably won't work in general. How should this be determined?

Reply:
> How should this be determined?

For source it should ( ref )  be obviously the source codec
h264_cuvid , hevc_cuvid or av1_cuvid

For encode ( dis ) it should always be av1_cuvid

Reply:
The vmaf command is designed to be used for arbitrary reference & distorted vids so the codec could be anything.

I think what we'll need is to detect the codec (from the ffprobe?) and map to a appropriate cude codec. Is it only h264, hevc & av1 supported?

Reply:
codec cant be anything on encode ( dis ) its av1_cuvid unless im missing the code where you can pick what codec ab-av1 outputs ?
for source ( ref ) this is what a modern nvidia supports :
av1_cuvid
h264_cuvid
hevc_cuvid
mjpeg_cuvid
mpeg1_cuvid
mpeg2_cuvid
mpeg4_cuvid
vc1_cuvid
vp8_cuvid
vp9_cuvid

Reply:
> we'll need is to detect the codec (from the ffprobe?)  

and yes on source ( ref ) you have to detect it manually

Reply:
> unless im missing the code where you can pick what codec ab-av1 outputs

`--encoder`

Reply:
> --encoder

welp then you will need to map all encoders to cuvid , the easy ones are _nvenc => _cuvid

Reply:
Ok I've added basic detection and setting